src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -477,21 +500,37 @@ public void OnResponseAbort(Exception abortException)
                 }
             }
 
+            private void CheckIfDisposedOrAborted()
+            {
+                Debug.Assert(Monitor.IsEntered(SyncObject));
+
+                if (_disposed)
+                {
+                    throw new ObjectDisposedException(nameof(Http2Stream));
+                }
+
+                if (_state == StreamState.Aborted)
+                {
+                    if (_canRetry)
+                    {
+                        // Throw a retryable request exception.
+                        // This will cause retry logic to kick in and perform another connection attempt.
+                        // The user should never see this exception. 
+                        throw new HttpRequestException(null, null, allowRetry: true);
```

Comment:
Nit: I think this is the third place now where we throw this same exception construct in order to force a retry.  Consider putting it behind some function that makes its intent clear, e.g.
```C#
throw Http2Connection.Retry();
```
or something like that.

Reply:
done